src/middlewared/middlewared/plugins/mdns.py

```diff
@@ -14,6 +16,76 @@
 
 from middlewared.service import Service
 
+class mDNSDaemonThread(threading.Thread):
+    def __init__(self, *args, **kwargs):
+        super(mDNSDaemonThread, self).__init__()
+        self.setDaemon(True)
+        self.mdnsd_pidfile = "/var/run/mdnsd.pid"
+        self.mdnsd_piddir = "/var/run/"
+
+    def pidfile_exists(self):
+        return os.access(self.mdnsd_pidfile, os.F_OK)
+
+    def is_alive(self):
+        mdnsd_proc = None
+        alive = False
+
+        if not self.pidfile_exists():
+            return False
+
+        for proc in psutil.process_iter():
+            procdict = proc.as_dict(['name'])
+            procname = str(procdict['name'])
+
+            match = re.search("([^/]+)$", procname)
+            if not match:
+                continue
+
+            procname = match.group(0)
+            if procname == "mdnsd":
+                mdnsd_proc = proc
+                break
```

Comment:
It's hard to understand what you're trying to achieve here. Is it `procname.split("/")[0] == "mdnsd"`?

Reply:
What is difficult to understand? It's a very simple regular expression.